onnxruntime/core/xnnpack/conv.cc

```diff
@@ -0,0 +1,299 @@
+  has_const_output_shape_ = IsAllDimKnown(output_shape_);
+
+  ORT_ENFORCE(info.TryGetConstantInput(1, &weight));
+  ORT_ENFORCE(info.TryGetConstantInput(2, &B));
+  const TensorShape& kernel_shape = weight->Shape();
+  cpu_allocator_ = info.GetAllocator(0, OrtMemTypeDefault);
+  weight_ = static_cast<float*>(cpu_allocator_->AllocArray(kernel_shape.Size(), sizeof(float)));
+  ORT_ENFORCE(weight_ != nullptr);
+  TensorShape new_weight_shape{kernel_shape[3], kernel_shape[1], kernel_shape[2], 1};
+  hwc_to_chw(weight->Data<float>(), kernel_shape[1], kernel_shape[2], kernel_shape[3], weight_);
+
+  int64_t input_channels = input_type_proto->tensor_type().shape().dim(3).dim_value();
+  // Weight shape : [ 1, kernel_height, kernel_width, input_channels * depth_multiplier ]
+  ORT_ENFORCE(kernel_shape.NumDimensions() == 4);
+  ORT_ENFORCE(kernel_shape[3] % input_channels == 0);
+  int64_t kernel_height = kernel_shape[1];
+  int64_t kernel_width = kernel_shape[2];
+
+  int64_t input_padding_top;
+  int64_t input_padding_right;
+  int64_t input_padding_bottom;
+  int64_t input_padding_left;
+
+  int64_t subsampling_height;
+  int64_t subsampling_width;
+  int64_t dilation_height;
+  int64_t dilation_width;
+  float output_min;
+  float output_max;
+  int64_t padding_mode;
+  ORT_ENFORCE(info.GetAttr("input_padding_top", &input_padding_top).IsOK());
+  ORT_ENFORCE(info.GetAttr("input_padding_right", &input_padding_right).IsOK());
+  ORT_ENFORCE(info.GetAttr("input_padding_bottom", &input_padding_bottom).IsOK());
+  ORT_ENFORCE(info.GetAttr("input_padding_left", &input_padding_left).IsOK());
+  ORT_ENFORCE(info.GetAttr("subsampling_height", &subsampling_height).IsOK());
+  ORT_ENFORCE(info.GetAttr("subsampling_width", &subsampling_width).IsOK());
+  ORT_ENFORCE(info.GetAttr("dilation_height", &dilation_height).IsOK());
+  ORT_ENFORCE(info.GetAttr("dilation_width", &dilation_width).IsOK());
+  // TODO: handle optional case
+  ORT_ENFORCE(info.GetAttr("output_min", &output_min).IsOK());
+  ORT_ENFORCE(info.GetAttr("output_max", &output_max).IsOK());
+  ORT_ENFORCE(info.GetAttr("padding_mode", &padding_mode).IsOK());
+
+  input_padding_top_ = gsl::narrow<uint32_t>(input_padding_top);
+  input_padding_right_ = gsl::narrow<uint32_t>(input_padding_right);
+  input_padding_bottom_ = gsl::narrow<uint32_t>(input_padding_bottom);
+  input_padding_left_ = gsl::narrow<uint32_t>(input_padding_left);
+  subsampling_height_ = gsl::narrow<uint32_t>(subsampling_height);
+  subsampling_width_ = gsl::narrow<uint32_t>(subsampling_width);
+  dilation_height_ = gsl::narrow<uint32_t>(dilation_height);
+  dilation_width_ = gsl::narrow<uint32_t>(dilation_width);
+  padding_mode_ = gsl::narrow<uint32_t>(padding_mode);
+
+  uint32_t flags = 0;
+  if (padding_mode == 1) {
+    flags |= XNN_FLAG_TENSORFLOW_SAME_PADDING;
+  }
+  int64_t depth_multiplier = kernel_shape[3] / input_channels;
+  struct xnn_operator* p;
+  xnn_status status = xnn_create_convolution2d_nhwc_f32(
```

Comment:
>xnn_create_convolution2d_nhwc_f32 [](http://example.com/codeflow?start=22&length=33)

any reason that we want to split depthwise and normal conv? looks like they call same function to create the xnn_operator

Reply:
The schema def is based on xnnpack's high-level graph API, so there is  a directly mapping between such an ONNX graph(built upon XNNPack Ops) and a xnnpack graph.  Their high-level graph API is more stable than the lower-level operator API(i.e. xnn_create_convolution2d_nhwc_f32). which is used for implementing.  The lower-level could get deprecated soon.  We'd prefer to keep the schemas stable. So it makes more sense to use the high-level API for defining the OP schemas.